mavros_extras/src/plugins/companion_process_status.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @brief Companion Status plugin
+ * @file companion_status.cpp
+ * @author Tanja Baumann <tanja@auterion.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2018 Tanja Baumann.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.mdA
+ */
+
+#include <mavros/mavros_plugin.h>
+
+#include <mavros_msgs/CompanionProcessStatus.h>
+
+namespace mavros {
+namespace extra_plugins {
+
+//! Mavlink enumerations
+using mavlink::common::MAV_TYPE;
+using mavlink::common::MAV_STATE;
+using utils::enum_value;
+
+/**
+ * @brief Obstacle companion process status plugin
+ *
+ * Publishes the status of components running on the companion computer
+ * @see status_cb()
+ */
+class CompanionProcessStatusPlugin : public plugin::PluginBase {
+public:
+	CompanionProcessStatusPlugin() : PluginBase(),
+	status_nh("~companion_process")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		status_sub = status_nh.subscribe("status", 10, &CompanionProcessStatusPlugin::status_cb, this);
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return { /* Rx disabled */ };
+	}
+
+private:
+	ros::NodeHandle status_nh;
+	ros::Subscriber status_sub;
+
+	/**
+	 * @brief Send companion process status to FCU
+	 *
+	 * Message specification: http://mavlink.org/messages/common#COMPANION_PROCESS_STATUS
+	 * @param req	received CompanionProcessStatus msg
+	 */
+
+
+
```

Comment:
Also can you please add doxygen type docs for the functions? Example: https://github.com/mavlink/mavros/blob/master/mavros_extras/src/plugins/trajectory.cpp#L194-L199

Reply:
@TSC21 I think in the plugin they already exist. where would you want those added?